### PR TITLE
Auto-gen Achievements interop layer

### PR DESCRIPTION
**Pull request by rakhad**
An example of the work required to upgrade an existing wrapper layer on top of the newly generated interop layer.

### CODE REVIEW

File: Package/Assets/GDK-APIs/Runtime/Source/Interop/XBL/Achievements/Methods.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        [DllImport(ThunkDllName, CallingConvention = CallingConvention.Cdecl, ExactSpelling = true)]
+        [return: NativeTypeName("HRESULT")]
+        public static extern int XblAchievementsResultDuplicateHandle([NativeTypeName("XblAchievementsResultHandle")] XblAchievementsResult* handle, [NativeTypeName("XblAchievementsResultHandle *")] XblAchievementsResult** duplicatedHandle);
+
+        [DllImport(ThunkDllName, CallingConvention = CallingConvention.Cdecl, ExactSpelling = true)]
+        public static extern void XblAchievementsResultCloseHandle([NativeTypeName("XblAchievementsResultHandle")] XblAchievementsResult* handle);
+
+        //[DllImport(ThunkDllName, CallingConvention = CallingConvention.Cdecl, ExactSpelling = true)]
+        //[return: NativeTypeName("XblFunctionContext")]
+        //public static extern int XblAchievementUnlockAddNotificationHandler([NativeTypeName("XblContextHandle")] XblContext* xblContextHandle, [NativeTypeName("XblAchievementUnlockHandler *")] delegate* unmanaged[Cdecl]<XblAchievementUnlockEvent*, void*, void>* handler, void* handlerContext);
```
Comment (rakhad) [inline Package/Assets/GDK-APIs/Runtime/Source/Interop/XBL/Achievements/Methods.cs:58]: >static [](http://example.com/codeflow?start=17&length=6)

Still investigating a solution to this.

Comment (rakhad) [inline Package/Assets/GDK-APIs/Runtime/Source/Interop/XBL/Achievements/Methods.cs:58]: > delegate* unmanaged[Cdecl]<XblAchievementUnlockEvent*, void*, void>* handler [](http://example.com/codeflow?start=194&length=77)

Still investigating a solution for this.

**Review by payzer — APPROVED** [review]

**Comment by mike-copley-atg** [comment]
So, is this still fit to merge in?  I am wanting to continue work where Ramsay left off and this PR looks like it has stuff that we want to have but it failed a build check?  Any thoughts @Vicodex or @payzer ?  thanks!

**Comment by payzer** [comment]
It would be good to get the build fixed. Do we know why it failed?

**Comment by mike-copley-atg** [comment]
Looks like it is trying to run a Unity build of some sort:

***@***.***

... and is perhaps missing a needed parameter or something.

Michael Copley
Senior Software Engineer
Microsoft Gaming Developer Advanced Technology Group

**Comment by payzer** [comment]
I'm not familiar with that error. But perhaps @Vicodex has an idea.

**Comment by payzer** [comment]
I'm going to close this PR.

**Pull request closed without merging**